### PR TITLE
Fixup cert package tests

### PR DESCRIPTION
**Pull request by nbrownus**
This change fixes the old v1 tests and adds new tests for v2. 

Most notably it fixes the `TestMarshalingCertificateV1Consistency` test which needed to be modified. `netip.Prefix` can not represent netmasks like `255.0.255.0`. I went back to f22b4b5 and modified the test details to remove the non conforming netmasks and adjust the timestamps, the resulting expected byte arrays are represented in this test.

There are two spots where I am unsure if we want to panic, v1 and v2 `String()` calls `json.Marshal` which can fail. Currently they return a string but panic would be more obvious.

### CODE REVIEW

File: cert/test_helpers.go [hunk 1]
```diff
@@ -1,28 +1,50 @@
-package e2e
+package cert
+
+//TODO: this should only be built if performing tests
```
Comment (nbrownus) [inline cert/test_helpers.go:3]: I really dont want to require we use a tag to execute tests but it also feels bad to include this in the binary and I don't know of another way to deal with this without just copying these functions all over the place again.

Comment (JackDoanRivian) [reply cert/test_helpers.go:3]: can't you define these in a `something_helpers_test.go` and use them in other `*_test.go` files?

Comment (nbrownus) [reply cert/test_helpers.go:3]: I desperately want there to be a way but the only thing I can come up with is a build tag much like how `e2e` works

**Comment by johnmaguire** [comment]
Could knock out https://github.com/slackhq/nebula/issues/1124 in this too.

**Pull request merged**